libs/s25main/lua/LuaPlayer.cpp

```diff
@@ -259,15 +261,22 @@ bool LuaPlayer::AIConstructionOrder(unsigned x, unsigned y, lua::SafeEnum<Buildi
     return true;
 }
 
+void LuaPlayer::PlaceHQ(MapCoord x, MapCoord y)
+{
+    // Ignore if there is an HQ set in the map file.
```

Comment:
This doesn't actually care about the map file directly. A script might (attempt to) set the HQ multiple times. Or re-add it if destroyed.

```suggestion
    // Ignore if an HQ already exists
```